### DISCUSSION TITLE
Tutorial won't compile: Meet The Composable Architecture - Testing presentation, Testing the add contact flow, Step 12 has an error

### DISCUSSION THREAD

**Discussion by takenori-kabeya** [pointfreeco/swift-composable-architecture#3530] [General]
I have encountered an error during the tutorial.

```swift
await store.send(\.destination.addContact.setName, "Blob Jr.") {
    $0.destination?.addContact?.contact.name = "Blob Jr."
    // Error: Setter for 'subscript(dynamicMember:)' is unavailable: Write 'enum = .case(value)', not 'enum.case = value'
}
```
I think the tip in the left side note corresponds exactly to this problem, but I don't know how to solve it.

>Tip
> To chain into an enum and mutate an associated value, the enum must be annotated with @CasePathable and @dynamicMemberLookup. The @Reducer macro automatically applies these annotations to enum-based State, but you must manually apply it to other enums.

I wrote Desitination enum as follows:

```swift
extension ContactsFeature {
    @Reducer
    enum Destination {
        case addContact(AddContactFeature)
        case alert(AlertState<ContactsFeature.Action.Alert>)
    }
}

extension ContactsFeature.Destination.State: Equatable {}
```

When expanding this @Reducer macro, it seems that State marked with @CasePathable and @dynamicMemberLookup is indeed generated.

Once I wrote it like this, it compiled up to Step 18 and all the tests passed.

```swift
await store.send(\.destination.addContact.setName, "Blob Jr.") {
    //$0.destination?.addContact?.contact.name = "Blob Jr."
            
    var addContact = $0.destination?.addContact
    addContact?.contact.name = "Blob Jr."
    $0.destination = .addContact(addContact!)
}
```

Is there anything I should check?

swift-composable-architecture version: 1.17.0
swift-case-paths version: 1.5.6
Xcode version: 16.2
iOS version: 18.1

**Comment by takenori-kabeya** [comment]
I don't know if the following line is necessary or not, but if I remove this line I am able to compile.

CasePathable.swift
```
@available(*, unavailable, message: "Write 'enum = .case(value)', not 'enum.case = value'")
```

**Comment by Butanediol** [comment]
I found this issue. https://github.com/pointfreeco/swift-composable-architecture/discussions/3158

```swift
await store.send(\.destination.addContact.setName, "Blob Jr.") {
    $0.destination?.modify(\.addContact) {
        $0.contact.name = "Blob Jr."
    }
}
```

**Reply by takenori-kabeya** [reply]
This seems to be exactly what is causing this. Thanks!
